packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -0,0 +1,3450 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ***********************************************************************
+//@HEADER
+*/
+
+#ifndef IFPACK2_BLOCKTRIDICONTAINER_IMPL_HPP
+#define IFPACK2_BLOCKTRIDICONTAINER_IMPL_HPP
+
+#include <Teuchos_Details_MpiTypeTraits.hpp>
+
+#include <Tpetra_Distributor.hpp>
+#include <Tpetra_BlockMultiVector.hpp>
+
+#include <Kokkos_ArithTraits.hpp>
+#include <KokkosBatched_Util.hpp>
+#include <KokkosBatched_Vector.hpp>
+#include <KokkosBatched_Copy_Decl.hpp>
+#include <KokkosBatched_Copy_Impl.hpp>
+#include <KokkosBatched_AddRadial_Decl.hpp>
+#include <KokkosBatched_AddRadial_Impl.hpp>
+#include <KokkosBatched_Gemm_Decl.hpp>
+#include <KokkosBatched_Gemm_Serial_Impl.hpp>
+#include <KokkosBatched_Gemm_Team_Impl.hpp>
+#include <KokkosBatched_Gemv_Decl.hpp>
+#include <KokkosBatched_Gemv_Serial_Impl.hpp>
+#include <KokkosBatched_Gemv_Team_Impl.hpp>
+#include <KokkosBatched_Trsm_Decl.hpp>
+#include <KokkosBatched_Trsm_Serial_Impl.hpp>
+#include <KokkosBatched_Trsm_Team_Impl.hpp>
+#include <KokkosBatched_Trsv_Decl.hpp>
+#include <KokkosBatched_Trsv_Serial_Impl.hpp>
+#include <KokkosBatched_Trsv_Team_Impl.hpp>
+#include <KokkosBatched_LU_Decl.hpp>
+#include <KokkosBatched_LU_Serial_Impl.hpp>
+#include <KokkosBatched_LU_Team_Impl.hpp>
+
+#include <memory>
+
+// need to interface this into cmake variable (or only use this flag when it is necessary)
+//#define BLOCKTRIDICONTAINER_ENABLE_PROFILE
+#undef  BLOCKTRIDICONTAINER_ENABLE_PROFILE
```

Comment:
Please namespace macros.  For example: `IFPACK2_BLOCKTRIDCONTAINER_ENABLE_PROFILE`.